code/modules/clothing/belts/polymorph_belt.dm

```diff
@@ -0,0 +1,152 @@
+/// Belt which can turn you into a beast, once an anomaly core is inserted
+/obj/item/polymorph_belt
+	name = "polymorphic field inverter"
+	desc = "This device can scan and store DNA from other life forms."
+	slot_flags = ITEM_SLOT_BELT
+	icon = 'icons/obj/clothing/belts.dmi'
+	icon_state = "polybelt_inactive"
+	worn_icon_state = "polybelt_inactive"
+	item_flags = NOBLUDGEON
+	/// Typepath of a mob we have scanned, we only store one at a time
+	var/stored_mob_type
+	/// Have we activated the belt?
+	var/active = FALSE
+	/// Our current transformation action
+	var/datum/action/cooldown/spell/shapeshift/polymorph_belt/transform_action
+
+/obj/item/polymorph_belt/Destroy(force)
+	QDEL_NULL(transform_action)
+	return ..()
+
+/obj/item/polymorph_belt/examine(mob/user)
+	. = ..()
+	if (stored_mob_type)
+		var/mob/living/will_become = stored_mob_type
+		. += span_notice("It contains digitised [initial(will_become.name)] DNA.")
+	if (!active)
+		. += span_warning("It requires a Bioscrambler Anomaly Core in order to function.")
+
+/obj/item/polymorph_belt/item_action_slot_check(slot, mob/user, datum/action/action)
+	return slot & ITEM_SLOT_BELT
+
+/obj/item/polymorph_belt/update_icon_state()
+	icon_state = active ? "polybelt" : "polybelt_inactive"
+	worn_icon_state = active ? "polybelt" : "polybelt_inactive"
+	return ..()
+
+/obj/item/polymorph_belt/attackby(obj/item/weapon, mob/user, params)
+	if (!istype(weapon, /obj/item/assembly/signaler/anomaly/bioscrambler))
+		return ..()
+	balloon_alert(user, "inserting...")
+	if (!do_after(user, delay = 3 SECONDS, target = src))
+		return
+	qdel(weapon)
+	active = TRUE
+	update_appearance(UPDATE_ICON_STATE)
+	user.update_worn_belt()
+	update_transform_action()
+	playsound(src, 'sound/machines/crate_open.ogg', 50, FALSE)
+
+/obj/item/polymorph_belt/attack(mob/living/target_mob, mob/living/user, params)
+	. = ..()
+	if (.)
+		return
+	if (!isliving(target_mob))
+		return
+	if (ishuman(target_mob) && !ismonkey(target_mob))
+		balloon_alert(user, "target too complex!")
+		return
+	if (!(target_mob.mob_biotypes & MOB_ORGANIC))
+		balloon_alert(user, "organic life only!")
+		return
+	if (isanimal_or_basicmob(target_mob))
+		if (!target_mob.compare_sentience_type(SENTIENCE_ORGANIC))
+			balloon_alert(user, "target too intelligent!")
+			return
```

Comment:
Does this prevent Megafauna from being scanned

Reply:
yes